app/exec/extension/install.ts

```diff
@@ -158,9 +160,20 @@ export class ExtensionInstall extends extBase.ExtensionBase<ExtensionInstallResu
 				}
 
 				trace.debug("Installation User Id:" + data.authenticatedUser.id);
-				return this._getAccountsByMemberId(data.authenticatedUser.id);
+
+				// Check if we're targeting the public Marketplace
+				if (data.locationServiceData && data.locationServiceData.serviceOwner == MARKETPLACE_INSTANCE_TYPE) {
+					trace.debug("Targeting public Marketplace. Prepare to get accounts for current authenticated user");
+					return this._getAccountsByMemberId(data.authenticatedUser.id).then(accounts => accounts.map(a => <TargetAccount>{ accountId: a.accountId, accountName: a.accountName }));
+				}
+				else {
+					// Assumning a local gallery. Let's read collections
+					trace.debug("Targeting a local gallery. Prepare to get project collections.");
+					const coreApi = this.webApi.getQCoreApi(this.webApi.serverUrl);
+					return coreApi.getProjectCollections().then(collections => collections.map(c => <TargetAccount>{ accountId: c.id, accountName: c.name }));
```

Comment:
I assume you've confirmed this API works properly on-prem? And what happens if it fails because a lack of permissions by the caller? I vaguely recall there being a pretty high bar for calling this API on-prem (e.g. deployment admin), but could be wrong.